Lib/sqlite3/test/dump.py

```diff
@@ -49,6 +50,67 @@ def test_table_dump(self):
         [self.assertEqual(expected_sqls[i], actual_sqls[i])
             for i in range(len(expected_sqls))]
 
+    def test_dump_autoincrement(self):
+        expected_sqls = [
+            """CREATE TABLE "posts" (id int primary key);"""
+            ,
+            """INSERT INTO "posts" VALUES(0);"""
+            ,
+            "CREATE TABLE \"tags\" ( " \
+            "id integer primary key autoincrement," \
+            "tag varchar(256)," \
+            "post int references posts);"
+            ,
+            """INSERT INTO "tags" VALUES(NULL,'test',0);"""
+            ,
+            "CREATE TABLE \"tags2\" ( " \
+            "id integer primary key autoincrement," \
+            "tag varchar(256)," \
+            "post int references posts);"
+        ]
+        [self.cu.execute(s) for s in expected_sqls]
+        i = self.cx.iterdump()
+        actual_sqls = [s for s in i]
+        # the NULL value should now be automatically be set to 1
+        expected_sqls[3] = expected_sqls[3].replace("NULL", "1")
+        expected_sqls = ['BEGIN TRANSACTION;'] + expected_sqls + \
+                        ['DELETE FROM "sqlite_sequence";'] + \
+                        ["""INSERT INTO "sqlite_sequence" VALUES('tags',1);"""] + \
+                        ['COMMIT;']
+        [self.assertEqual(expected_sqls[i], actual_sqls[i])
+         for i in range(len(expected_sqls))]
```

Comment:
Please format the new code according to [PEP 8](https://www.python.org/dev/peps/pep-0008/). Even if large parts of these files _do not_ follow PEP 8, we strive to follow PEP 8 (and PEP 7) when adding new code, or modifying existing code :)